### PR TITLE
feat: Lu improvements, larger Lg editor, activity action node content

### LINKED ISSUE #6413: [LU Editor] Add toolbar for define/use entities
Designs: https://www.figma.com/file/q76DseJ7PK0g7zI1gXLzlD/SH_Language-Generation?node-id=1655%3A45665

- [x] Add toolbar to LU editor (define/use entity)
- [x] Add define entity button to menu, ML entity vs. prebuilt entity with search
- [x] Add use entity button to menu: use entity when text selected (only ML entity) vs. not (+ prebuilt)
- [x] Add labeling context menu to selected text in Monaco

### LINKED ISSUE #6502: [LG Editor Improvement] Allow editing lg templates in a larger editor
Design: https://www.figma.com/file/q76DseJ7PK0g7zI1gXLzlD/SH_Language-Generation?node-id=40%3A3

- [ ] Add switch to larger editor button for lg code editor (code view)
- [ ] Add switch to larger editor button for lg code editor in attachment variations
- [ ] Show larger lg code editor in a portal when user requests larger editor

### LINKED ISSUE #6503: [Action Node] Alternative display text when node has structured response
Design:
https://www.figma.com/file/q76DseJ7PK0g7zI1gXLzlD/SH_Language-Generation?node-id=40%3A3

- [ ] Change displayed text on LG nodes when they have structured response
- [ ] Update node header according to design when they have structured response

### PR DESCRIPTION
**Pull request by hatpick**
## Description
- Improvement to the Lu editor by adding a toolbar for defining/inserting ML/prebuilt entities
- Adds floating labeling menu for ML entities to Lu editor
- Adds larger Lg editor for when in code editing mode or attachment editing in response editor.
- Custom action node content for Lg templates with structured response.

## Task Item

closes #6503 
closes #6502 
closes #6413 

### Action node custom text for structured responses
Action node
### Lu editor improvements
Lu toolbar
### Floating labeling menu
Floating labeling menu
### Lu toolbar - insert entity
insert entity
### Lu toolbar - define entity
Define entity
### Lg response editor - attachment larger editor
Attachment larger editor
RE - Larger editor
### Lg code editor - larger editor
Lg code editor - larger editor
CE - Larger editor

### CODE REVIEW

File: Composer/packages/client/src/pages/language-understanding/code-editor.tsx [hunk 1]
```diff
@@ -153,18 +153,20 @@ const CodeEditor: React.FC<CodeEditorProps> = (props) => {
...
         }}
+        luFile={file}
         luOption={luOption}
         value={content}
         onChange={onChange}
         onChangeSettings={handleSettingsChange}
       />
     );
-  }, [luOption]);
+  }, [luOption, file]);
```
Comment (GeoffCoxMSFT) [inline Composer/packages/client/src/pages/language-understanding/code-editor.tsx:163]: Need to add the onChange callback as dependency

File: Composer/packages/lib/code-editor/src/hooks/useSearchableMenuListCallback.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            placeholder={searchFiledPlaceHolder ?? formatMessage('Search ...')}
+            styles={searchFieldStyles}
+            onAbort={onSearchAbort}
+            onChange={onSearchQueryChange}
+          />
+          <Stack styles={itemsContainerStyles}>{defaultRender?.(menuListProps)}</Stack>
+        </Stack>
+      );
+    },
+    [headerRenderer, onSearchAbort, onSearchQueryChange]
```
Comment (GeoffCoxMSFT) [inline Composer/packages/lib/code-editor/src/hooks/useSearchableMenuListCallback.tsx:45]: also searchFiledPlaceHolder ?

File: Composer/packages/lib/code-editor/src/lg/LgCodeEditor.tsx [hunk 3]
```diff
@@ -170,37 +177,86 @@ export const LgCodeEditor = (props: LgCodeEditorProps) => {
...
+
+  const onExpandedEditorChange = React.useCallback(
+    (newValue: string) => {
+      editor?.getModel()?.setValue(newValue);
+      onChange(newValue);
+    },
+    [editor, onChange]
+  );
+
+  const change = React.useCallback(
```
Comment (GeoffCoxMSFT) [inline Composer/packages/lib/code-editor/src/lg/LgCodeEditor.tsx:191]: Nit: naming - handleChange?

File: Composer/packages/lib/code-editor/src/lg/modalityEditors/StringArrayEditor.tsx [hunk 4]
```diff
@@ -118,7 +118,11 @@ export const StringArrayEditor = React.memo(
 
     useEffect(() => {
       const keydownHandler = (e: KeyboardEvent) => {
-        if (submitKeys.includes(e.key)) {
+        if (
+          e.key === 'Enter' ||
```
Comment (GeoffCoxMSFT) [inline Composer/packages/lib/code-editor/src/lg/modalityEditors/StringArrayEditor.tsx:122]: Are there no defined constants for keys to compare with?  (C# has KeyCodes, but I'm not sure about React)

Comment (hatpick) [reply Composer/packages/lib/code-editor/src/lg/modalityEditors/StringArrayEditor.tsx:122]: key"codes" are deprecated
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

File: Composer/packages/types/src/indexers.ts [hunk 5]
```diff
@@ -105,6 +105,7 @@ export enum LuSectionTypes {
 
 export type LuEntity = {
   Name: string;
```
Comment (GeoffCoxMSFT) [inline Composer/packages/types/src/indexers.ts:107]: Uppercase property names?

Comment (hatpick) [reply Composer/packages/types/src/indexers.ts:107]: existing code

File: Composer/packages/ui-plugins/luis/src/LuisIntentEditor.tsx [hunk 6]
```diff
@@ -40,6 +40,19 @@ const LuisIntentEditor: React.FC<FieldProps<string>> = (props) => {
...
+    (luFileId: string, sectionId?: string) => {
+      // eslint-disable-next-line security/detect-non-literal-regexp
+      const pattern = new RegExp(`.${locale}`, 'g');
+      const fileId = currentDialog.isFormDialog ? luFileId : luFileId.replace(pattern, '');
+      const url = currentDialog.isFormDialog
+        ? `/bot/${projectId}/language-understanding/${currentDialog.id}/item/${fileId}`
+        : `/bot/${projectId}/language-understanding/${fileId}${sectionId ? `/edit?t=${sectionId}` : ''}`;
+      shellApi.navigateTo(url);
+    },
+    [shellApi, projectId, locale]
```
Comment (GeoffCoxMSFT) [inline Composer/packages/ui-plugins/luis/src/LuisIntentEditor.tsx:53]: Also currentDialog ?

**Review by GeoffCoxMSFT — APPROVED** [review]

**Review by GeoffCoxMSFT — APPROVED** [review]

**Pull request merged**